static/main.js

```diff
@@ -5,6 +5,14 @@
 import katex from '../katex.webpack.js';
 import './main.css';
 
+function argumentRegex(keyRegex) {
+    return new RegExp(`(?:^\\?|&)(${keyRegex})=([^&]*)`, "g");
+}
+
+function argument(keyRegex) {
+    return argumentRegex(keyRegex).exec(window.location.search);
+}
```

Comment:
Would it be possible to document functions?

Reply:
I would probably refactor things to parse all the URL params to an object using a library from npm.  Then we could have simpler regexes for just the keys to check if there are any macros and the rest become simple key lookups.

Reply:
This has turned out to simplify the code a lot.  Good idea, @kevinbarabash!